### PR TITLE
Update documentation and db configuration attributes

### PR DESCRIPTION
**Pull request by twom**
The main work was to update the documentation to better reflect the way forward where the `mongodb[:config]` attribute contains the values written to the mongodb.conf file.

The code also contains cleanup to consistently use this new way of setting the mongodb configuration options.

Related issues are #222 and #182

### CODE REVIEW

**Comment by jamesonjlee** [comment]
looks clean enough for me, I will look into merging this as the start of the 0.15.X

**Comment by twom** [comment]
@jamesonjlee  thx. I did not change the Changelog file, but there could be a mention there to make sure people get that setting the configuration changed in the last versions.

**Comment by jamesonjlee** [comment]
@twom, could you update this for the rubocop changes?
it should be safe to just copy over the `.rubocop.yml`, run `(bundle exec) rubocop --auto-correct` and then rebase on top of master.

**Comment by twom** [comment]
@jamesonjlee ok, wil do. You are referring to a rebase on top of the edelight:master branch or just my master branch (the last case would not result in anything).

**Comment by jamesonjlee** [comment]
top of edelight, (and to deal with any conflicts, you can update yours by running rubocop on it first).

**Comment by twom** [comment]
@jamesonjlee I did as requested. Let me know if you need any other changes to be made

File: README.md [hunk 1]
```diff
@@ -72,8 +87,7 @@ For examples see the USAGE section below.
 
 # USAGE:
 
-## 10gen repository
-
+## 10gen
```
Comment (jamesonjlee) [inline README.md:90]: never call them MongoDB haha..ha....

**Comment by jamesonjlee** [comment]
this :lgtm, I will merge pending the change with configsrv fix (last 0.14 release), and this will mark 0.15.0 release. thanks @twom

File: attributes/dbconfig.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # if node.platform_family?("rhel", "fedora") then
 case node['platform_family']
 when 'rhel', 'fedora'
-  default['mongodb']['config']['fork'] = true
+    default['mongodb']['config']['fork'] = true
```
Comment (jspiewak) [inline attributes/dbconfig.rb:16]: did an extra indent level get added?

Comment (jamesonjlee) [inline attributes/dbconfig.rb:16]: looks like it.

**Pull request closed without merging**